File: lib/layer2_core/light/e_inverse_on_surface_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EInverseOnSurfaceLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl8;
      case 'standard': return cstl8;
      case 'blue': return cbl8;
      case 'green': return cgl8;
      case 'red': return crl8;
      case 'nothing': return cnol8;
      default: return cstl8;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EInverseOnSurfaceLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EInverseOnSurfaceLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl8
            AppTheme.STANDARD -> R.color.cstl8
            AppTheme.BLUE -> R.color.cbl8
            AppTheme.GREEN -> R.color.cgl8
            AppTheme.RED -> R.color.crl8
            AppTheme.NOTHING -> R.color.cnol8
        }
    }
}
